lib/rich-text/delta.rb

```diff
@@ -452,14 +452,16 @@ def to_json(*args)
     # The behavior is not defined with non-insert operations.
     # @param embed_str [String] the string to use in place of non-string insert operations
     # @return [String]
-    def to_plaintext(embed_str: '!')
-      @ops.each_with_object('') do |op, str|
+    def to_plaintext
+      plaintext = @ops.each_with_object('') do |op, str|
         if op.insert?(String)
           str << op.value
-        elsif embed_str
-          str << embed_str
+        elsif block_given?
+          val = yield(op)
+          str << val if val.is_a?(String)
         end
       end
+      plaintext.strip.gsub(/[\n\r]+/, "\n")
```

Comment:
What's with this post-processing step? Seems like it would coalesce legitimate multi-line breaks into one.

Reply:
Omitted objects leave lots of holes (and subsequent line breaks) in the document. You're right, this may be a better implementation detail for the consumer app.